lib/pbench/test/unit/server/test_user_auth.py

```diff
@@ -408,11 +410,16 @@ def test_forgot_password_valid_email(client, server_config):
             )
             assert resp_register.status_code == HTTPStatus.CREATED
 
+        def ok(self, msg: str):
+            return
+
+        monkeypatch.setattr(Mail, "send", ok)
```

Comment:
Just as a comment, you could have avoided duplicating this code by creating a `@pytest.fixture` doing this, and referencing the fixture in both tests. That's not a big deal, however.

Reply:
The mock function should be setting a flag defined in the test so that the test can check whether the mock was called:
```
    def test_forgot_password_valid_email(client, monkeypatch, server_config):
        mail_sent = False
        with client:
            ...

        def send_mock(self, msg: str):
            mail_sent = True

        monkeypatch.setattr(Mail, "send", send_mock)

        resp_forgot_password = forgot_password(client, server_config, "user@domain.com")
        assert resp_forgot_password.status_code == HTTPStatus.OK
        assert mail_sent
```
(This is a little awkward to do if the mock is set up in a fixture.)

You should use the same approach in `test_forgot_password_invalid_email()`, except that the additional assertion checks that the mail was _not_ sent.